h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -382,11 +402,56 @@ public static Vec categoricalDomainsToNumeric(final Vec src) {
       }.doAll(Vec.T_NUM, src).outputFrame().anyVec();
   }
 
+  public static class CollectDoubleDomain extends MRTask<CollectDoubleDomain> {
```

Comment:
That's true, it would save a lot for constant/binary chunks/CUD chunks. I am not sure if it's worth adding extra function on chunk though. 

Using sparse interface is definitely a good idea though. Good catch.